tools/acrn-crashlog/acrnprobe/README.rst

```diff
@@ -0,0 +1,176 @@
+system, ``acrnprobe`` provides ``history_event`` (under ``/var/log/crashlog/``
+by default) to manage the crash and events records on the platform instead of
+``telem_journal``. But in this case, the records can't be delivered to the
+backend.
+
+Usage
+*****
+
+The ``acrnprobe`` is launched as a service at boot. Also, it provides some basic
+options:
+
+Specify a configuration file for ``acrnprobe``. If this option is unused,
+``acrnprobe`` will use the configuration file located in CUSTOM CONFIGURATION
+PATH or INSTALLATION PATH (see `CONFIGURATION FILES`_).
+
+.. code-block:: console
+
+   $ acrnprobe -c [configuration_path]
+
+To see the version of ``acrnprobe``.
+
+.. code-block:: console
+
+   $ acrnprobe -V
+
+Architecture
+************
+
+Syntax
+======
+
+- channel :
+  Channel represents a way of detecting the system's events. There are 3
+  channels:
+
+  + oneshot: detect once while ``acrnprobe`` startup.
+  + polling: run a detecting job with fixed time interval.
+  + inotify: monitor the change of file or dir.
+
+- event queue :
+  There is a global queue to receive all events detected.
+  Generally, events are enqueued in channel, and dequeued in event handler.
+
+- event handler :
+  Event handler is a thread to handle events detected by channel.
+  It's awakened by an enqueued event.
+
+- sender :
+  The sender corresponds to an exit of event.
+  There are two senders:
+
+  + Crashlog is responsible for collecting logs and saving it locally.
+  + Telemd is responsible for sending log records to telemetrics client.
+
+Description
+===========
+
+As a log collection mechanism to record critical events on the platform,
+``acrnprobe`` provides these functions:
+
+1. detect event
+
+   From experience, the occurrence of an system event is usually accompanied
+   by some effects. The effects could be a generated file, an error message in
+   kernel's log, or a system reboot. To get these effects, for some of them we
+   can monitor a directory, for other of them we might need to do a detection
+   in a time loop.
+   *So we implement the channel, which represents a common method of detection.*
+
+2. analyze event and determine the event type
+
+   Generally, a specific effect correspond to a particular type of events.
+   However, it is the icing on the cake for analyzing the detailed event types
+   according to some phenomena. *Crash reclassify is implemented for this
+   purpose.*
+
+3. collect information for detected events
+
+   This is for debug purpose. Events without information are meaningless,
+   and developers need to use this information to improve their system. *Sender
+   crashlog is implemented for this purpose.*
+
+4. archive these information as logs, and generate records
+
+   There must be a central place to tell user what happened in system.
+   *Sender telemd is implemented for this purpose.*
+
+Diagram
+=======
+::
+
+ +---------------------------------------------+
```

Comment:
You have it right for the other diagrams.  Add two spaces before each line in this diagram.